pandas/core/reshape/merge.py

```diff
@@ -1263,8 +1263,8 @@ def _get_merge_keys(self):
             else:
                 lt = left_join_keys[-1]
 
-            msg = "incompatible tolerance, must be compat " \
-                  "with type {0}".format(type(lt))
+            msg = "incompatible tolerance {0}, must be compat " \
+                  "with type {1}".format(type(tolerance), lt.dtype)
```

Comment:
And I suppose it has to be `self.tolerance` instead of `tolerance` (see failing tests)